homeassistant/core.py

```diff
@@ -1168,9 +1168,9 @@ def __init__(
         self.parent_id = parent_id
         self.origin_event: Event[Any] | None = None
 
-    def __eq__(self, other: Any) -> bool:
+    def __eq__(self, other: object) -> bool:
         """Compare contexts."""
-        return bool(self.__class__ == other.__class__ and self.id == other.id)
+        return bool(type(other) is Context and self.id == other.id)
```

Comment:
```suggestion
        return isinstance(other, Context) and self.id == other.id
```
I'd suggest to use something like this.